rust-src/concordium_base/src/id/id_prover.rs

```diff
@@ -207,19 +221,68 @@ pub fn prove_ownership_of_account(
 /// that lower <= attribute < upper.
 /// This is done by proving that attribute-upper+2^n and attribute-lower lie in
 /// [0, 2^n). For further details about this technique, see page 15 in <https://arxiv.org/pdf/1907.06381.pdf>.
+#[allow(clippy::too_many_arguments)]
 pub fn prove_attribute_in_range<C: Curve, AttributeType: Attribute<C::Scalar>>(
+    version: &ProofVersion,
+    transcript: &mut RandomOracle,
+    csprng: &mut impl rand::Rng,
     gens: &Generators<C>,
     keys: &PedersenKey<C>,
     attribute: &AttributeType,
     lower: &AttributeType,
     upper: &AttributeType,
     r: &PedersenRandomness<C>,
 ) -> Option<RangeProof<C>> {
-    let mut transcript = RandomOracle::domain("attribute_range_proof");
-    let mut csprng = rand::thread_rng();
     let delta = attribute.to_field_element();
     let a = lower.to_field_element();
     let b = upper.to_field_element();
+    match version {
+        ProofVersion::Version1 => {
+            let mut transcript_v1 = RandomOracle::domain("attribute_range_proof");
+            prove_attribute_in_range_helper(
+                &ProofVersion::Version1,
+                &mut transcript_v1,
+                csprng,
+                gens,
+                keys,
+                delta,
+                a,
+                b,
+                r,
+            )
+        }
+        ProofVersion::Version2 => {
+            transcript.add_bytes(b"AttributeRangeProof");
+            transcript.append_message(b"a", &a);
+            transcript.append_message(b"b", &b);
+            prove_attribute_in_range_helper(
+                &ProofVersion::Version2,
+                transcript,
+                csprng,
+                gens,
+                keys,
+                delta,
+                a,
+                b,
+                r,
+            )
+        }
+    }
+}
+
+/// Helper function for producing a range proof.
+#[allow(clippy::too_many_arguments)]
+fn prove_attribute_in_range_helper<C: Curve>(
+    version: &ProofVersion,
```

Comment:
Please don't take references to `ProofVersion`. This is a very small value so references are both inconvenient, and pointless.